server/src/main/java/org/apache/iotdb/db/query/executor/groupby/SlidingWindowGroupByExecutorFactory.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.query.executor.groupby;
+
+import org.apache.iotdb.db.qp.constant.SQLConstant;
+import org.apache.iotdb.db.query.executor.groupby.impl.EmptyQueueSlidingWindowGroupByExecutor;
+import org.apache.iotdb.db.query.executor.groupby.impl.MonotonicQueueSlidingWindowGroupByExecutor;
+import org.apache.iotdb.db.query.executor.groupby.impl.NormalQueueSlidingWindowGroupByExecutor;
+import org.apache.iotdb.db.query.executor.groupby.impl.SmoothQueueSlidingWindowGroupByExecutor;
+import org.apache.iotdb.tsfile.exception.write.UnSupportedDataTypeException;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+
+public class SlidingWindowGroupByExecutorFactory {
+
+  public static SlidingWindowGroupByExecutor getSlidingWindowGroupByExecutor(
+      String aggrFuncName, TSDataType dataType, boolean ascending) {
+    if (aggrFuncName == null) {
+      throw new IllegalArgumentException("AggregateFunction Name must not be null");
+    }
+
+    switch (aggrFuncName.toLowerCase()) {
+      case SQLConstant.SUM:
+      case SQLConstant.AVG:
+      case SQLConstant.COUNT:
+        return new SmoothQueueSlidingWindowGroupByExecutor(dataType, aggrFuncName, ascending);
+      case SQLConstant.MAX_VALUE:
+        return new MonotonicQueueSlidingWindowGroupByExecutor(
```

Comment:
I'm not sure the what the performance impaction is of the boxing/unboxing. Is there any other way to avoid them?
Maybe we can use primitive type comparsion directly according to the input dataType?

Reply:
@JackieTien97 What do you think?

Reply:
We can add `createComparator` method in `TSDataType` to save the boxing and unboxing overhead.

Reply:
Unfortunately, `TSDataType` is in `tsfile` module and `AggregateResult` is in `server` module, so we cannot add a method in `TSDataType` to return `Comparator<AggregateResult>`. 

The best way is to move all these common class into a `common` module, but that has nothing to do with this pr, so you can try another way that is add `private static final Comparator<AggregateResult> comparators[] = new Comparator[6];` in `SlidingWindowGroupByExecutorFactory` and use `comparators[dataType.ordinal()]` to get corresponding comparator.

Reply:
> Unfortunately, `TSDataType` is in `tsfile` module and `AggregateResult` is in `server` module, so we cannot add a method in `TSDataType` to return `Comparator<AggregateResult>`.
> 
> The best way is to move all these common class into a `common` module, but that has nothing to do with this pr, so you can try another way that is add `private static final Comparator<AggregateResult> comparators[] = new Comparator[6];` in `SlidingWindowGroupByExecutorFactory` and use `comparators[dataType.ordinal()]` to get corresponding comparator.

If so, just use a `Map<TSDataType, Comparator>` instead of indexing by enum's `ordinal()`, which may depend on the definition order.

Reply:
fixed~